build-libs/integration-packer-redirects.js

```diff
@@ -0,0 +1,895 @@
+// The Slugs of the Integration Component Types
+const TYPE_BUILDER = 'builder'
+const TYPE_PROVISIONER = 'provisioner'
+const TYPE_POST_PROCESSOR = 'post-processor'
+const TYPE_DATA_SOURCE = 'data-source'
+
+// Takes a Component Type slug and returns the old Plugin Library equivalent
+function getOldPluginComponentSlug(componentType) {
+	switch (componentType) {
+		case TYPE_BUILDER:
+			return 'builders'
+		case TYPE_PROVISIONER:
+			return 'provisioners'
+		case TYPE_POST_PROCESSOR:
+			return 'post-processors'
+		case TYPE_DATA_SOURCE:
+			return 'datasources'
+		default:
+			return ''
+	}
+}
+
+/**
+ * An array of legacy Packer plugins that were previously in the
+ * Packer Plugin library that are now in the Integrations library.
```

Comment:
If there's a migration guide/document that we can link to in this comment, that would be great.

It's not immediately clear if this list is expected to grow or if it's stable, or it needs to change in tandem with some external work.

Reply:
☝️ related question: this is related to https://github.com/hashicorp/packer/blob/main/website/data/plugins-manifest.json right? Essentially redirecting the plugins detailed in that manifest to their new integration locations?